### ISSUE TITLE
FaceServiceClient.DetectAsync throws NullReferenceException

### ISSUE THREAD

**Issue by preulo** [Azure-Samples/storage-queue-dotnet-pop-receipt#1]
I downloaded this sample, updated the packages and followed the instructions to run it using the Storage Emulator. For every JPG file used, a NullReferenceException is throw at line 147:

var faces = await faceClient.DetectAsync(fileStream, false, true, new FaceAttributeType[] { FaceAttributeType.Age });

Here is the exception info:

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.ProjectOxford.Face.FaceServiceClient.<SendRequestAsync>d__67`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ProjectOxford.Face.FaceServiceClient.<DetectAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at PopreceiptSample.Program.<ProcessImages>d__2.MoveNext() in C:\Users\cmartins\Downloads\storage-queue-dotnet-pop-receipt-master\dotnet\storage-queue-dotnet-popreceipt\Program.cs:line 147
```

I also attached the files I am using to test the face detection feature, the whole Project with the packages updates and screenshots showing my system information.
[IssueAttachments.zip](https://github.com/Azure-Samples/storage-queue-dotnet-pop-receipt/files/3207365/IssueAttachments.zip)

**Comment by preulo** [comment]
I did forget to add the Face API key to the .config file. But, even after adding that info, the error persists.

**Comment by preulo** [comment]
When I noticed every operation of the FaceServiceClient was throwing the same exception, I suspected the problem could be the sample itself: it seems to be outdated. Eventually I found out a way to avoid the problem.

The constructor used for FaceServiceClient in this sample uses only the subscription key as a parameter:

`FaceServiceClient faceClient = new FaceServiceClient(CloudConfigurationManager.GetSetting("FaceAPIKey"));`

What I did was use the other constructor, which also accepts the apiRoot as a parameter:

`FaceServiceClient faceClient = new FaceServiceClient("my face api key", "https://region.api.cognitive.microsoft.com/face/v1.0");`

Using this constructor , the sample runs smoothly.
